data/shortcuts.json

```diff
@@ -158,6 +158,10 @@
             "shortcuts": ["}", "⇟"],
             "text": "shortcuts.browsing.vertex_selected.last"
           },
+          {
+            "shortcuts": ["<"],
```

Comment:
(Repeating this comment in a thread so it’s easier to discuss.)

I couldn’t think of an intuitive keyboard shortcut that was as convenient as the <kbd>[</kbd> and <kbd>]</kbd> shortcuts that it would complement. So I went with <kbd><</kbd>, based on (of all things) the [<code><</code> operator](https://wiki.openstreetmap.org/wiki/Overpass_API/Overpass_QL#Recurse_down_.28.3E.29) in OverpassQL, which recurses up to the parent of a node or way. I’m open to other ideas.

<kbd><</kbd> is a common character on keyboard layouts due to its prevalence in basic math. Just to be on the safe side, I consulted [Apple’s visual guide to physical keyboard localizations](https://support.apple.com/en-us/HT201794) and checked each of the 169 keyboard layouts available on macOS:

* Hindi Devanagari, Khmer, Myanmar, Sinhala, and Tibetan Wylie lack a <kbd><</kbd> key, though I didn’t check whether there are any dead keys that provide access to this character.
* Chinese Pinyin, Chinese Shuangpin, and Thai lack a <kbd><</kbd> key, but Thai keyboards have a <kbd><</kbd>-labeled key, so I suspect the expectation is that users would switch to a QWERTY keyboard when necessary.

To accommodate these keyboard layouts, we could make the keyboard shortcut localizable, relying on an assumption that keyboard layouts generally correspond to languages. If we choose a letter as the keyboard shortcut instead, then it would definitely have to be localizable.

All the other keyboard layouts have a <kbd><</kbd> key, but some require pressing both Shift and AltGr at the same time, which can be inconvenient.

Reply:
I originally went with punctuation because it seems like iD generally reserves letters for editing operations and uses control keys or punctuation for navigation, which makes sense. The problem is that there are very few control keys left that aren’t used by iD or the browser, and very few punctuation characters are universal among keyboard layouts.

An alternative I didn’t consider earlier was <kbd>Ctrl</kbd><kbd>↑</kbd> on Windows or ⌘↑ on macOS. These keys are physical keys on every keyboard, regardless of locale.

This combination is unlikely to cause a conflict with browsers. On Windows, <kbd>Ctrl</kbd><kbd>↑</kbd> would move the caret to the beginning of the paragraph, but this shortcut would take effect when no text field has focus. On macOS, ⌘↑ would move the caret to the beginning of a text field, and it’s also Finder’s shortcut for moving up to the containing folder.

Reply:
> but iD doesn’t scroll like a page

If the focus is somehow on the sidebar (but not one of its fields), then pressing ⌘↓ jumps to the bottom of the scroll view but ⌘↑ would select the parent ways instead of jumping to the top of the scroll view. I think this situation would be relatively rare, but hopefully there’s a way to special-case the shortcut to only occur when the canvas has focus. Otherwise, that would be a good argument for making the shortcut <kbd>Ctrl</kbd><kbd>Shift</kbd><kbd>↑</kbd> on Windows and ⇧⌘↑ on macOS.